rfcs/0000-upgradeable-lifecycle.md

```diff
@@ -0,0 +1,76 @@
+**Problem:**
+Currently, the lifecycle version used in kpack is not configurable, nor is it user upgradeable. Users must wait for a new kpack release to be able to upgrade the version of the lifecycle used by their builders.
+
+**Outcome:**
+We would like to decouple the lifecycle upgrade process from the kpack upgrade process, allowing users to upgrade the lifecycle on their own.
+
+**Actions to take:**
+There are 2 major paths that we can go down to provide this functionality, each with their own pros and cons.
+
+1. A simpler, monolithic approach that would keep a single lifecycle version for the whole cluster like it is today, but with the functionality fo configuring that version exposed. The main benefit of soing something like that it will likely be the least intrusive on user workflow. The lifecycle would continue to be something that is more back of mind for users. There are multiple ways to handle this approach:
+	1. Create a command in `kp` (i.e `kp lifecycle upgrade`) that would give users the ability to manually update their lifecycle
+	2. Provide a script similiar to our `hack/lifecycle.go` that users can run 
+	3. Add the concept of lifecycle version to our descriptor files so that `kp import` can upgrade the lifecycle.  
+
+2. A more complex approach that would treat the lifecycle as a resource like a store or stack. Lifecycles would be tied to each builder so that different builders can have different lifecycle versions.This has the benefit of allowing users to test new lifecycle versions without having to spin up another cluster. Also, users who created their own buildpack would not have to worry about a newer lifecycle dropping support for the Buildpack Api they are using. 
```

Comment:
In a user interview with a kpack users at Nasdaq he stated that the currently assign a cluster per application.  Just food for thought that it may not be burdensome to utilize many clusters.

Reply:
Good point about custom buildpack creation.  If we go with the option of creating a lifecycle resource, this would allow App Ops (primary buildpack author persona imo) to not need to ask a Platform Operator to upgrade the lifecycle.  However, if we support upgrading via the descriptor file and customers are `kp import`ing frequently, seems like this wouldn't be a problem.